### ISSUE TITLE
Android build Failed JS sourcemap file doesn't exist

### ISSUE THREAD

**Issue by herparis** [DataDog/dd-sdk-reactnative#511]
**Describe what happened**

Android Build is failing due to not finding the sourcemap file

```
Script '${PROJECT}/node_modules/@datadog/mobile-react-native/datadog-sourcemaps.gradle' line: 87

JS sourcemap file doesn't exist, aborting upload
```

**Steps to reproduce the issue:**
- Add react-native sourcemap config
- Try to build android

**Describe what you expected:**
- Android must build successfully

**Additional context**

-   Version of the SDK: 31
-   Version of React Native: 0.71.4
-   `package.json`
-   `Podfile` and `Podfile.lock`
-   `android/build.gradle` and `android/app/build.gradle`

**Comment by herparis** [comment]
ISSUE FOUND ON:
node_modules/@datadog/mobile-react-native/datadog-sourcemaps.gradle' line: 52

`def jsSourceMapsDir = file("$buildDir/generated/sourcemaps/react/${targetPath}")`

targetPath=`${variant}/${buildType}` example='dev/debug'

sourcemaps is being stored on the next path:

`$buildDir/generated/sourcemaps/react/devDebug`

following the described above, if we use `targetPath`, the path would be instead:

`$buildDir/generated/sourcemaps/react/dev/debug`

so it will be going to have issues trying to find the `index.android.bundle.map` file

SOLUTION:

change `node_modules/@datadog/mobile-react-native/datadog-sourcemaps.gradle'` line: 52 from:

`def jsSourceMapsDir = file("$buildDir/generated/sourcemaps/react/${targetPath}")`

to:

`def jsSourceMapsDir = file("$buildDir/generated/sourcemaps/react/${variant.name}")`

**Comment by louiszawadzki** [comment]
Hi @herparis, thanks for reaching out! 

Which version of `@datadog/mobile-react-native` are you using?
From what I see [in the current version](https://github.com/DataDog/dd-sdk-reactnative/blob/develop/packages/core/datadog-sourcemaps.gradle#L31) `targetPath` is being set to `variant.dirName`, not `${variant}/${buildType}`.
Could you try to upgrade `@datadog/mobile-react-native` to use this change?

We often have to make updates to support new versions of React Native as some things around sourcemaps are not public APIs and can change from version to version, so it's often a good idea to upgrade our SDK when upgrading React Native :) 

Let me know if that fixed your issue!

**Comment by bennettp123** [comment]
I just ran into this issue using `@datadog/mobile-react-native@1.8.3`—can confirm that the issue still occurs with `@datadog/mobile-react-native@1.8.4`, which is currently the latest version, and using `variant.dirName` to set `targetPath`.

I think this is the issue—the source map does not exist at `variant.dirName`, but it does exist at `variant.name`. If I add some extra debug logs to the error handler:

```
targetPath is dev/release
variant.dirName is dev/release
variant.name is devRelease
jsSourceMapsDir is /Users/bennett/src/west-app/android/app/build/generated/sourcemaps/react/dev/release
jsOutputSourceMapFile is /Users/bennett/src/west-app/android/app/build/generated/sourcemaps/react/dev/release/index.android.bundle.map
```

`jsOutputSourceMapFile` indeed refers to a file that does not exist—should be `devRelease/index.android.bundle.map`, not `dev/release/index.android.bundle.map`

edit: raised #528 with additional context (repro steps, gradle files, etc)

**Comment by louiszawadzki** [comment]
Hi @herparis, 

We released version `1.8.5` of the SDK that contains a fix for your problem. 
I'll close the issue for now, feel free to reopen it if the issue isn't fixed!
